src/test/java/org/jitsi/meet/test/base/AbstractTest.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright @ 2015 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.meet.test.base;
+
+import org.openqa.selenium.*;
+import org.testng.*;
+import org.testng.annotations.*;
+
+import java.lang.reflect.*;
+import java.util.*;
+import java.util.stream.*;
+
+import static org.jitsi.meet.test.util.TestUtils.*;
+
+/**
+ * The base of all tests, checks whether the tests is supposed to be executed
+ * or skipped by checking the properties which tests to run, exclude or include.
+ *
+ * @author Damyan Minkov
+ * @author George Politis
+ */
+@Listeners(FailureListener.class)
+public abstract class AbstractTest
```

Comment:
so will we have AbstractTest and AbstractBaseTest classes ? can we come up with better naming, so that it would indicate what's the difference between the two ? Otherwise we could as well name them AbstractBaseTest1 and AbstractBaseTest2...

Reply:
Thanks for the comments Pawel, I'll try to come up with a better name for the new class.

The bwe estimation tests do need participant management (if you've spotted a place where it says differently please point that out to me and I'll fix it), but the lifecycle is different from what we have now: In the bwe estimation tests the test method is executed in parallel and each parallel test method launches 2 participants. Does that shed some light?

Reply:
Yes, it does shed some lights. I don't like how it duplicates the lifecycle procedure. There must be another way to make it parallel. I'll try to think about it.

Reply:
So one way would be to compose a class which would extend WebTestBase. Then inside the "test"/"runScenario" method you can manually set it up by passing the Properties to it's "setupClassPrivate". After the test is done just call it's cleanup methods.

Because of that duplication you have already skipped one step from the initialisation phase:
https://github.com/jitsi/jitsi-meet-torture/blob/master/src/test/java/org/jitsi/meet/test/base/AbstractBaseTest.java#L133